conformance/lang/expressions/list_constructor.balt

```diff
@@ -0,0 +1,1242 @@
+Test-Case: parser-error
+Description: Test list constructor negative syntax.
+Labels: list-constructor-expr, array-type
+
+function errorFunction() {
+    int[] arr1 = [; // @error missing close bracket token
+    int[] arr2 = ]; // @error missing open bracket token
+    int[] arr3 = [
+        1
+        2 // @error missing comma token
+        3 // @error missing comma token
+    ];
+}
+
+Test-Case: output
+Description: Test 1 for list-constructor-expr occurring within a const-expr will construct a structural value
+             that has its read-only bit on.
+Fail-Issue: ballerina-platform/ballerina-lang#13944
+Labels: module-const-decl, list-constructor-expr, is-expr, array-type, readonly-type, string
+
+const string[] strArr = [];
+const int[] intArr = [1, 2, 3, 4];
+
+function init() {
+    io:println(strArr is readonly); // @output true
+    io:println(intArr is readonly); // @output true
+}
+
+Test-Case: output
+Description: Test 2 for list-constructor-expr occurring within a const-expr will construct a structural value
+             that has its read-only bit on.
+Fail-Issue: ballerina-platform/ballerina-lang#13944
+Labels: module-const-decl, list-constructor-expr, is-expr, tuple-type, readonly-type, string, boolean
+
+const [int, string, boolean] tuple1 = [11, "foo", true];
+const [int, [boolean, string], boolean] tuple2 = [11, [true, "foo"], true];
+
+function init() {
+    io:println(tuple1 is readonly); // @output true
+    io:println(tuple2 is readonly); // @output true
+}
+
```

Comment:
Shall we add a test to check during list construction each exp is being evaluated in order?
> The members of the list come from evaluating each expression in the expr-list in order.

```ballerina
int[] arr = [foo(), foo()]; // foo returns incremented value per each invocation
```